2.4.41/Dockerfile

```diff
@@ -0,0 +1,47 @@
+FROM centos:centos7
+
+# OpenLDAP server image for OpenShift Origin
+#
+# Volumes:
+# * /var/lib/ldap/data - Datastore for OpenLDAP
+# * /etc/openldap/     - Config directory for slapd
+# Environment:
+# * $OPENLDAP_ADMIN_PASSWORD         - OpenLDAP administrator password
+# * $OPENLDAP_DEBUG_LEVEL (Optional) - OpenLDAP debugging level, defaults to 256
+
+MAINTAINER Steve Kuznetsov <skuznets@redhat.com>
+
+ENV OPENLDAP_VERSION=2.4.41 \
+    HOME=/var/lib/openldap
+
+LABEL io.k8s.description="OpenLDAP is an open source implementation of the Lightweight Directory Access Protocol." \
+      io.k8s.display-name="OpenLDAP 2.4.41" \
+      io.openshift.expose-services="389:ldap,636:ldaps" \
+      io.openshift.tags="directory,ldap,openldap,openldap2441" \
+      io.openshift.non-scalable="true"
+
+# Install OpenLDAP Server
+RUN yum install -y openldap openldap-servers openldap-clients && \
+    yum clean all -y && \
+    mkdir -p /var/lib/ldap && \
+    mkdir -p /etc/openldap && \
+    chmod a+rw -R /var/lib/ldap && \
+    chmod a+rw -R /etc/openldap
+
+# Set OpenLDAP data and config directories in a data volume
+VOLUME ["/var/lib/ldap", "/etc/openldap"]
```

Comment:
config on (persistent) volume is better so you can actually modify it and have it stick around through a pod restart.  that's the pattern i used for the jenkins image.

otherwise you have to build a new image to bake your config into it, and that's a burden, we don't to make users create their own layer on top of the image just to configure it.

if we were going to go that route, then this should be an s2i image where the "config" is your source-repo and the "Assemble" script applies that config.  (There certainly are advantages to having an immutable, pre-configured, image, so we might need to discuss expected usage a bit more.  In the case of jenkins i'd expect people to be frequently changing the config, so baking it into the image felt wrong)